docs/DeveloperGuide.md

```diff
@@ -1,5 +1,166 @@
+beginner software engineers will be able to levitate their expertises and mindset in the process of developing 
+and presenting a new product.
+
+### Maintainability
+With the aim of increasing maintainability of Fluffle, separated packages and classes were implemented in strict 
+compliance with Object-oriented Programming. Since it is important to be able to easily change functionality of one 
+class without disturbing other dependent classes, each class only serves one purpose to reduce dependency on other 
+classes. With low coupling and high cohesion, subsequent developers can make minimal effort in maintaining Fluffle.
+
+## Definitions
+
+|Terminology|Explanation|
+|--------|----------|
+|Plot bunny|A story idea that refuses to go away until it is written down.|
+|Fluffle|A fluffle is a group of bunnies.| 
+|CLI|Command-line Interface. </br></br>A command-line interface (CLI) processes command to a computer program in the form of lines of text. The program in the computer will read the command and start proceeding.|
+|Gradle|A build tool used for automated testing. It checks for code style violations and runs unit tests to ensure the code is functional between iterations.| 
+|IntelliJ|An integrated development environment that used to write and test java code.|
+
+## Setting up the project in your computer
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
+
+<div markdown="span" class="alert alert-warning">
+
+⚠ **Caution:**
+
+Follow the steps in the following guide precisely. Things will not work out if you deviate in some steps.
+</div>
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.<br>
+  :exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**:
+   1. Run the `java -jar duke.jar` and try a few commands.
+   2. [Run the tests](Testing.md) to ensure they all pass. 
+//Note: should ask team W11-02 the permission for the above part.
+
+## System Overview
+
+### Technological view
+The Integrated Development Environment, IntelliJ, is used to develop our program. The program is written in Java, 
+and uses Gradle for building and testing purposes. Our source code is mostly original, with some functions imported 
+from the java.util package. The remaining packages and classes which form the structure of our program 
+are independently developed.
+
+## Architecture
+![UML Fluffle class diagram](graphics/diagrams/classDiagram_Fluffle.png)
```

Comment:
The diagram is too big and unclear, as it is an overview of the whole program, what if you split it into smaller sections so that it is easier to understand. Maybe you could split according to different packages. I have to zoom into the diagram to see what is there.
<img width="928" alt="Screenshot 2020-10-28 at 10 42 32 AM" src="https://user-images.githubusercontent.com/49362047/97395485-13a75480-190b-11eb-9fa1-2a5f0f8cef56.png">